lib/sshutils/scp/scp.go

```diff
@@ -683,12 +778,16 @@ type Destination struct {
 // 3 components of it
 func ParseSCPDestination(s string) (*Destination, error) {
 	out := reSCP.FindStringSubmatch(s)
-	if len(out) == 0 {
+	if len(out) != 4 {
```

Comment:
Are you sure it's always going to have exactly 4 matching groups? Looking at the regex it seems like some groups are optional (username in particular).